File: longest-palindromic-subsequence/longest-palindromic-subsequence.cpp
class Solution {
public:
    int longestPalindromeSubseq(string s) {
        string t = s;
        reverse(t.begin( ) , t.end());
        int n=s.length();
        vector< vector<int>> dp(n+1 , vector<int>(n+1,0));
        
        for(int i=1;i<n+1;i++)
        {
            for(int j=1;j<n+1;j++)
            {
                if(s[i-1]==t[j-1])
                    dp[i][j] = 1+dp[i-1][j-1];
                else
                    dp[i][j] = max(dp[i-1][j] , dp[i][j-1]);
            }
        }
        return dp[n][n];
    }
};

File: longest-palindromic-subsequence/longest-palindromic-subsequence.py
class Solution:
    def longestPalindromeSubseq(self, s: str) -> int:
        n = len(s)
        dp = [[0 for _ in range(n+1) ] for _ in range(n+1)]
        for i in range(1,n+1):
            for j in range(1,n+1):
                if( s[i-1] == s[n-j]):
                    dp[i][j] = 1+dp[i-1][j-1]
                else:
                    dp[i][j] = max(dp[i-1][j],dp[i][j-1])
        return dp[n][n]
       
        